test/CadlRanchProjects.Tests/type-property-optionality.cs

```diff
@@ -385,7 +385,7 @@ public class TypePropertyOptionalTests : CadlRanchTestBase
         public Task Type_Property_Optional_UnionFloatLiteral_getAll() => Test(async (host) =>
         {
             Response response = await new OptionalClient(host, null).GetUnionFloatLiteralClient().GetAllAsync(new RequestContext());
-            BinaryData.Equals(BinaryData.FromObjectAsJson(2.375), UnionFloatLiteralProperty.FromResponse(response).Property);
+            Assert.AreEqual(BinaryData.FromObjectAsJson(2.375).ToString(), UnionFloatLiteralProperty.FromResponse(response).Property.ToString());
```

Comment:
Not the case. The `BinaryData` class does not have a `Equals` static method defined. Therefore this `BinaryData.Equals` is actually calling the `object.Equals` method, which compares the two argument by **reference**.
Please clarify, are we expecting the two BinaryData instance to be the same one? Or we are asserting "they have the same content".

Also I think we should use
```suggestion
            CollectionAssert.AreEqual(BinaryData.FromObjectAsJson(2.375).ToArray(), UnionFloatLiteralProperty.FromResponse(response).Property.ToArray());
```
because we are using this to assert the content of binarydata in the other PR

Reply:
@mcgallan please use this new method to assert BinaryData instances: https://github.com/Azure/autorest.csharp/pull/4570/files

Reply:
fixed